File: brady/binary_search.java
class Solution {
  
    // O(lg(n)) time, O(1) space
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1;
        int m;

        while (l <= r) {
            m = l+((r-l)/2);    // Avoids overflow
            if (nums[m] == target) return m;
            else if (nums[m] < target) l = m+1;
            else r = m-1;
        }
        return -1;
    }
}

File: gabe/binary_search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int h = nums.size()-1;
        int m = (l+h)/2;

        while(l <= h) {
            if(target < nums[m]) {
                h = m-1;
            } else if (target == nums[m]) {
                return m;
            } else {
                l = m + 1;
            }

            m = (l+h)/2;
        }
        return -1;
    }
};